engine/src/main/java/org/terasology/utilities/collection/DirectedAcyclicClassGraph.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.utilities.collection;
+
+import com.google.api.client.util.Maps;
+
+import java.util.Map;
+
+public class DirectedAcyclicClassGraph<T> extends DirectedAcyclicGraph<T> {
```

Comment:
I won't make comments about these two classes but I'd like to talk about them again before this PR gets merged. Primarily my question is, why -two- classes instead of a more typical interface/class combo? Time for me to go to bed!